sdk/cmake/CMakeMacros.cmake

```diff
@@ -683,6 +733,15 @@ function(set_module_type MODULE TYPE)
         set_target_properties(${MODULE} PROPERTIES SUFFIX ".cpl")
     endif()
 
+    # Set the Windows version being compiled for
+    if(TARGET_VERSION)
```

Comment:
Let's do it in a little bit different way:
- `TARGET_VERSION` should always be set
- The default value is "win2003"
- If some module wants to override `TARGET_VERSION`, it should be greater than the default one, otherwise default one is used. (you may use `list(FIND ...)` cmake feature for comparing "greatness")

Reply:
I believe you may be misunderstanding this functionality. An explicitly specified target version for the entire OS (using `-DTARGET_VERSION=` parameter in configure.sh/cmd) is checked first. If that is not set, then the module’s manually specified target version is used, if any. Otherwise, it defaults to `win2003`.

Reply:
Hm, yeah.
I wanted to say, the default value should be set in config.cmake. That way, you don't need to check for it in set_module_type